driver-core/src/main/com/mongodb/internal/operation/ChangeStreamBatchCursor.java

```diff
@@ -64,7 +64,12 @@ public boolean hasNext() {
         return resumeableOperation(new Function<AggregateResponseBatchCursor<RawBsonDocument>, Boolean>() {
             @Override
             public Boolean apply(final AggregateResponseBatchCursor<RawBsonDocument> queryBatchCursor) {
-                return queryBatchCursor.hasNext();
+                try {
+                    return queryBatchCursor.hasNext();
+                } catch (RuntimeException e) {
```

Comment:
It might also be ok for this to be a finally block, but I wasn't able to convince myself of it.

Reply:
I'm interested as to why do it this way? Personally, I'd convert to a finally block as its cleaner and in my head equates to the same thing.

Reply:
I convinced myself it's ok for this to be a finally block.